tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
             doc = html.fromstring(res.get_data(as_text=True))
             assert len(doc.xpath('//input[@id="input-PR1-no"]/@checked')) == 1
 
-    def test_post_valid_data(self, data_api_client):
+    def test_get_with_with_prefilled_answers(self, data_api_client):
         with self.app.test_client():
             self.login()
+            # First call is for the current framework; second call for the framework to pre-fill from
+            data_api_client.get_framework.side_effect = [
+                self.framework(slug='g-cloud-9', name='G-Cloud 9', status='open'),
+                self.framework(slug='digital-outcomes-and-specialists-2', name='Digital Stuff 2', status='live'),
+            ]
+            # Current framework application information
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                framework_slug="g-cloud-9",
+                declaration={"status": "started"},
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from
+            data_api_client.get_supplier_declaration.return_value = {
```

Comment:
What you're describing above isn't unit testing. 
A test with more condition checks may cover more code but they are inherently more fragile and add far less value by being less indicative of an overall problem.

Put it this way, we could check, in every one of our unit tests, whether the GOV.UK cookie banner appears.
Say we then add a check to every test to make sure the CrownCopyright appears in the bottom right.

Then, on the next test run, everything breaks. 
Was it the banner or the CC or both?

We don't do the above because it doesn't add value, every test would just break if we changed the banner.
Unless you're checking one specific thing unit test failures help _narrow down_ a problem at best and at worst point you completely in the wrong direction.

Obviously this is a massive oversimplification but it demonstrates why it's important to check only what you want to find out. Tests are cheap, once you've written one it's very easy to write more so if needs be then split them out into several tests that functionally do the same thing but check different conditions.

Then when a test breaks called `test_view_makes_only_one_api_call` fails you know _exactly_ what's wrong instead of a test called `test_declaration_page` failing because someone updated the frontend toolkit.